### PR TITLE
[ReferenceManager][Projects][2/n] - Projects Permissions, Create, Update & Sub Create

### PR DESCRIPTION
**Pull request by calvinhlee23**
- permssions get
- create api
- update api 
- sub project api

### CODE REVIEW

**Comment by le-sun** [comment]
@calvinhlee23 Tests are failing

**Comment by calvinhlee23** [comment]
hmm doesn't look related to my pr. i'll check

File: src/citation/views/citation_project_view.py [hunk 1]
```diff
@@ -1,13 +1,51 @@
...
+from user.related_models.organization_model import Organization
 
 
 class CitationProjectViewSet(ModelViewSet):
     queryset = CitationProject.objects.all()
     permission_classes = [IsAuthenticated]
     serializer_class = CitationProjectSerializer
     ordering = ["-updated_date"]
+
+    def list(self, request):
```
Comment (le-sun) [inline src/citation/views/citation_project_view.py:19]: We should make this as an `@action` endpoint, so we can take in an pk for organization

File: src/citation/views/citation_project_view.py [hunk 2]
```diff
@@ -1,13 +1,51 @@
...
+        try:
+            current_user = request.user
+            org_id = request.query_params.get("organization", None)
+            if org_id.endswith("/"):
+                org_id = org_id[:-1]
+            org = Organization.objects.get(id=int(org_id))
```
Comment (le-sun) [inline src/citation/views/citation_project_view.py:25]: Then we can just do `Organization.objects.get(id=pk)`

**Review by le-sun — CHANGES_REQUESTED** [review]

**Pull request merged**